AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/plugin/PluginClassHolder.java

```diff
@@ -59,4 +62,13 @@ protected AbstractJClass getJClass(Class<?> clazz) {
 	protected JCodeModel getCodeModel() {
 		return environment().getCodeModel();
 	}
+
+	protected ProcessingEnvironment getProcessingEnvironment() {
```

Comment:
Why would plugins need this? The plugin system is created in a way that plugins should only be called by handlers.

Reply:
@WonderCsabo , any plugin with "a little of complexity" could need to do stuff with the ProcessingEnvironment. Like getting the ElementUtils or TypeUtils to make any enhanced processing over elements. This is just a convenient method, the plugin always can call environment().getProcessingEvironment().. but I don't see the point in not facilitating this (in the same way that it was done with the getCodeModel()..

Reply:
You are right, sorry